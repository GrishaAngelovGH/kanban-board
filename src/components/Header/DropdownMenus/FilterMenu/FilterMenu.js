import Button from "react-bootstrap/Button"
import ButtonGroup from "react-bootstrap/ButtonGroup"
import Dropdown from "react-bootstrap/Dropdown"
import DropdownButton from "react-bootstrap/DropdownButton"
import FormControl from "react-bootstrap/FormControl"

import boardRepository from "persistent/persistentKanbanBoardRepository"

const FilterMenu = ({ onUpdate }) => {
  const handleFilterByPriority = ({ target: { innerText } }) => {
    boardRepository.applyPriorityFilter(innerText)
    onUpdate()
  }

  const handleSearch = ({ target }) => {
    boardRepository.applySearchFilter(target.value)
    onUpdate()
  }

  const handleClearFilters = () => {
    boardRepository.clearFilters()
    onUpdate()
  }


  const priority = boardRepository.getPriorityFilter()
  const searchCriteria = boardRepository.getSearchFilter() || ""
  const variant = priority || searchCriteria ? "success" : "light"

  return (
    <DropdownButton as={ButtonGroup} size="sm" variant={variant} title={<i className="bi bi-filter"></i>}>
      <Dropdown.ItemText className="bg-secondary-subtle text-secondary border text-center">Filter by priority</Dropdown.ItemText>
      {
        ["All", "Low", "Medium", "High"].map((item, i) => (
          <Dropdown.Item key={i} onClick={handleFilterByPriority} >
            {item} {item === priority && <i className="bi bi-check-circle-fill text-primary fs-5 ms-2"></i>}
          </Dropdown.Item>
        ))
      }

      <Dropdown.ItemText className="bg-secondary-subtle text-secondary border text-center">Filter by criteria</Dropdown.ItemText>
      <Dropdown.ItemText>
        <FormControl
          value={searchCriteria}
          placeholder="Search"
          className="mt-2 mb-2"
          onChange={handleSearch}
        />
      </Dropdown.ItemText>

      <Dropdown.ItemText className="bg-secondary-subtle text-secondary border text-center">Clear All Filters</Dropdown.ItemText>
      <Dropdown.Item>
        <Button variant="danger" size="sm" className="bi bi-x-circle w-100 mt-1" onClick={handleClearFilters}></Button>
      </Dropdown.Item>
    </DropdownButton>
  )
}

export default FilterMenu