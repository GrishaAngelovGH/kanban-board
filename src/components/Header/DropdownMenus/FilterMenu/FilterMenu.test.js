import { render, fireEvent, act, waitFor } from "@testing-library/react"

import FilterMenu from "./FilterMenu"

test("should render FilterMenu component", async () => {
  const view = render(<FilterMenu />)

  act(() => {
    const parentElement = view.container.firstChild
    const dropdownButton = parentElement.firstChild

    fireEvent.click(dropdownButton)
  })

  await waitFor(() => {
    expect(view).toMatchSnapshot()
  })
})