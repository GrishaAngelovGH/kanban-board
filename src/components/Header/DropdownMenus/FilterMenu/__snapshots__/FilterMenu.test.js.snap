// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render FilterMenu component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="show dropdown btn-group"
        role="group"
      >
        <button
          aria-expanded="true"
          class="dropdown-toggle show btn btn-light btn-sm"
          type="button"
        >
          <i
            class="bi bi-filter"
          />
        </button>
        <div
          aria-labelledby=""
          class="dropdown-menu show"
          x-placement="bottom-start"
        >
          <span
            class="bg-secondary-subtle text-secondary border text-center dropdown-item-text"
          >
            Filter by priority
          </span>
          <a
            class="dropdown-item"
            data-rr-ui-dropdown-item=""
            href="#"
            role="button"
            tabindex="0"
          >
            All
          </a>
          <a
            class="dropdown-item"
            data-rr-ui-dropdown-item=""
            href="#"
            role="button"
            tabindex="0"
          >
            Low
          </a>
          <a
            class="dropdown-item"
            data-rr-ui-dropdown-item=""
            href="#"
            role="button"
            tabindex="0"
          >
            Medium
          </a>
          <a
            class="dropdown-item"
            data-rr-ui-dropdown-item=""
            href="#"
            role="button"
            tabindex="0"
          >
            High
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="show dropdown btn-group"
      role="group"
    >
      <button
        aria-expanded="true"
        class="dropdown-toggle show btn btn-light btn-sm"
        type="button"
      >
        <i
          class="bi bi-filter"
        />
      </button>
      <div
        aria-labelledby=""
        class="dropdown-menu show"
        x-placement="bottom-start"
      >
        <span
          class="bg-secondary-subtle text-secondary border text-center dropdown-item-text"
        >
          Filter by priority
        </span>
        <a
          class="dropdown-item"
          data-rr-ui-dropdown-item=""
          href="#"
          role="button"
          tabindex="0"
        >
          All
        </a>
        <a
          class="dropdown-item"
          data-rr-ui-dropdown-item=""
          href="#"
          role="button"
          tabindex="0"
        >
          Low
        </a>
        <a
          class="dropdown-item"
          data-rr-ui-dropdown-item=""
          href="#"
          role="button"
          tabindex="0"
        >
          Medium
        </a>
        <a
          class="dropdown-item"
          data-rr-ui-dropdown-item=""
          href="#"
          role="button"
          tabindex="0"
        >
          High
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`should render FilterMenu component with priority 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="show dropdown btn-group"
        role="group"
      >
        <button
          aria-expanded="true"
          class="dropdown-toggle show btn btn-success btn-sm"
          type="button"
        >
          <i
            class="bi bi-filter"
          />
        </button>
        <div
          aria-labelledby=""
          class="dropdown-menu show"
          x-placement="bottom-start"
        >
          <span
            class="bg-secondary-subtle text-secondary border text-center dropdown-item-text"
          >
            Filter by priority
          </span>
          <a
            class="dropdown-item"
            data-rr-ui-dropdown-item=""
            href="#"
            role="button"
            tabindex="0"
          >
            All
          </a>
          <a
            class="dropdown-item"
            data-rr-ui-dropdown-item=""
            href="#"
            role="button"
            tabindex="0"
          >
            Low
          </a>
          <a
            class="dropdown-item"
            data-rr-ui-dropdown-item=""
            href="#"
            role="button"
            tabindex="0"
          >
            Medium
          </a>
          <a
            class="dropdown-item"
            data-rr-ui-dropdown-item=""
            href="#"
            role="button"
            tabindex="0"
          >
            High
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="show dropdown btn-group"
      role="group"
    >
      <button
        aria-expanded="true"
        class="dropdown-toggle show btn btn-success btn-sm"
        type="button"
      >
        <i
          class="bi bi-filter"
        />
      </button>
      <div
        aria-labelledby=""
        class="dropdown-menu show"
        x-placement="bottom-start"
      >
        <span
          class="bg-secondary-subtle text-secondary border text-center dropdown-item-text"
        >
          Filter by priority
        </span>
        <a
          class="dropdown-item"
          data-rr-ui-dropdown-item=""
          href="#"
          role="button"
          tabindex="0"
        >
          All
        </a>
        <a
          class="dropdown-item"
          data-rr-ui-dropdown-item=""
          href="#"
          role="button"
          tabindex="0"
        >
          Low
        </a>
        <a
          class="dropdown-item"
          data-rr-ui-dropdown-item=""
          href="#"
          role="button"
          tabindex="0"
        >
          Medium
        </a>
        <a
          class="dropdown-item"
          data-rr-ui-dropdown-item=""
          href="#"
          role="button"
          tabindex="0"
        >
          High
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
